security/providers/oidc/src/main/java/io/helidon/security/providers/oidc/OidcSupport.java

```diff
@@ -260,6 +264,44 @@ private void processLogout(ServerRequest req, ServerResponse res) {
                 .exceptionallyAccept(t -> sendError(res, t));
     }
 
+    private String findTenantName(ServerRequest request) {
+        List<String> missingLocations = new LinkedList<>();
+        Optional<String> tenantId = Optional.empty();
+        if (oidcConfig.useParam()) {
+            tenantId = request.queryParams().first(oidcConfig.tenantParamName());
+
+            if (tenantId.isEmpty()) {
+                missingLocations.add("query-param");
+            }
+        }
+        if (oidcConfig.useCookie() && tenantId.isEmpty()) {
+            tenantId = oidcConfig.tenantCookieHandler()
+                    .findCookie(request.headers().toMap())
+                    .stream()
+                    .map(this::getValueFromSingle)
+                    .findFirst();
+
+            if (tenantId.isEmpty()) {
+                missingLocations.add("cookie");
+            }
+        }
+        if (tenantId.isPresent()) {
+            return tenantId.get();
+        } else {
+            LOGGER.finest(() -> "Missing tenant id, could not find in either of: " + missingLocations);
```

Comment:
1. This should be guarded by `if (LOGGER.isEnabledFor(FINEST)` (pseudocode)
2. Use only a single log statement - does not make sense to call logger twice for a single message.

Reply:
OK, will fix that